packages/api/src/mappings/google/activity.ts

```diff
@@ -40,23 +58,51 @@ export const mapToActivity = (
 
         if (data.dataSourceId === sourceIdActiveMinutes) {
           const seconds = convert(sum(intValues)).from("min").to("s");
-
           activity.summary = {
             ...activity.summary,
             durations: {
               ...activity.summary?.durations,
               active_seconds: seconds,
             },
           };
+          activity.activity_logs = activity.activity_logs?.map(act => {
+            const matchingActivity = data.point.find(point =>
+              matchActivityTime(act.start_time, act.end_time, parseInt(point.startTimeNanos))
+            );
+            if (matchingActivity) {
+              act = {
+                ...act,
+                durations: {
+                  ...act.durations,
+                  active_seconds: calculateActiveSeconds(act.start_time, act.end_time),
+                },
+              };
+            }
+            return act;
+          });
         }
 
         if (data.dataSourceId === sourceIdCalories) {
           activity.summary = {
             ...activity.summary,
             energy_expenditure: {
-              active_kcal: sum(values),
+              active_kcal: formatNumber(sum(values)),
             },
           };
+          activity.activity_logs = activity.activity_logs?.map(act => {
+            const matchingActivity = data.point.find(point =>
+              matchActivityTime(act.start_time, act.end_time, parseInt(point.startTimeNanos))
+            );
+            if (matchingActivity && matchingActivity.value[0].fpVal) {
```

Comment:
what if there's >1 value in the array? or is this not possible?

Reply:
I haven't seen or been able to make it have more than 1 value